### PR TITLE
Use ops docker spec #2

### PR DESCRIPTION
**Pull request by jaredkerim**

### CODE REVIEW

**Comment by jaredkerim** [comment]
@mostlygeek @relud I've implemented parts of the docker spec found here: 

https://github.com/mozilla-services/Dockerflow

Logging will be handled by #7 but I think this should handle the other core components.  Does everything look okay to you?

File: embedly/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
-WORKDIR /embedly
-COPY . /embedly
-RUN pip install -r /embedly/requirements.txt
+
+RUN groupadd --gid 10001 app && \
+    useradd --uid 10001 --gid 10001 --home /app --create-home app
+
+WORKDIR /app
+COPY . /app
+RUN pip install -r /app/requirements.txt
```
Comment (relud) [inline embedly/Dockerfile]: I'm under the impression that caching goes better if you copy just requirements.txt into the container first, run pip install, and then do

**Comment by relud** [comment]
the rest lgtm

File: embedly/Dockerfile (hunk 1 above)
Comment (mostlygeek) [inline embedly/Dockerfile]: yes it better to install requirements first then copy in the source

**Comment by mostlygeek** [comment]
rest looks good to me

**Comment by mostlygeek** [comment]
re: circle-ci build failure. Since there is the `USER app` in the Dockerfile, pip will fail with the permission problem. If you had `--user root` to the `docker run` command it should work. 

for testing, I think it is OK to run as `root`.

File: embedly/gunicorn.conf [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import multiprocessing
 
+access_logfile = "-"
```
Comment (oyiptong) [inline embedly/gunicorn.conf]: we should actually not log... or rather, log to /dev/null for privacy reasons

Comment (oyiptong) [inline embedly/gunicorn.conf]: ... at least until we do POST

**Comment by oyiptong** [comment]
@mostlygeek, @relud, we need to drop access logs on the host machine as well as the ELB.

This is a leak of users' history until we implement POST API requests. Very PII.

File: embedly/gunicorn.conf (hunk 2 above)
Comment (jaredkerim) [inline embedly/gunicorn.conf]: @oyiptong Yep you are correct that will be handled by #29 where we can disable logging at the container level.  The services within the container will log to stdout and then the docker daemon will redirect that output to /dev/null

**Comment by relud** [comment]
I'll make sure this is set up to drop logs when I deploy it.

**Comment by oyiptong** [comment]
most excellent

File: embedly/gunicorn.conf (hunk 2 above)
Comment (relud) [inline embedly/gunicorn.conf]: I would prefer this to log to /dev/null if it should be dropped. It's easier for me to make sure nothing is kept here, than elsewhere.

Comment (relud) [inline embedly/gunicorn.conf]: but understandably, having this as stdout here may make troubleshooting possible in stage

Comment (jaredkerim) [inline embedly/gunicorn.conf]: @relud Well eventually (read: soon) we WILL monitoring the logging, it's just that right now we log certain things in the URL that can be used to reconstruct users histories.  Soon that will no longer be recorded in the URL but rather in a POST body and will want to log requests just for traffic monitoring and performance reasons.  I think by the time it reaches ops to deploy this should already be taken care of and you will just be able to monitor normally.  I will make it very clear in the docs for you to deploy.

Comment (relud) [inline embedly/gunicorn.conf]: +1

**Pull request merged**